test/GPv2Settlement.test.ts

```diff
@@ -5,14 +5,20 @@ import { ethers, waffle } from "hardhat";
 import { domain } from "../src/ts";
 
 describe("GPv2Settlement", () => {
+  const [owner, solver] = waffle.provider.getWallets();
```

Comment:
I believe `owner` also acts as a deployer for `GPv2Settlement`, while this won't be the case for the real contracts.

You could make the deployer explicit with:
```
const [deployer, owner, solver] = waffle.provider.getWallets();
```
And then (instead of `.connect`) use the [optional signer argument of `ethers.getContractFactory`](
https://www.npmjs.com/package/@nomiclabs/hardhat-ethers#helpers) to make the deployer address immediately visible when creating the factory:
```
const GPv2SimpleAuthentication = await ethers.getContractFactory("GPv2SimpleAuthentication", owner);
const GPv2Settlement = await ethers.getContractFactory("GPv2SettlementTestInterface", deployer);
```

Reply:
Great suggestions!